site_scons/buildutils.py

```diff
@@ -34,6 +36,409 @@
+                **************************************************
+
+        The following options can be passed to SCons to customize the Cantera
+        build process. They should be given in the form:
+
+            scons build option1=value1 option2=value2
+
+        Variables set in this way will be stored in the 'cantera.conf' file and reused
+        automatically on subsequent invocations of scons. Alternatively, the
+        configuration options can be entered directly into 'cantera.conf' before
+        running 'scons build'. The format of this file is:
+
+            option1 = 'value1'
+            option2 = 'value2'
+        """
+        )
+    ]
+
+    def __init__(self):
+        self.options = {}
+        self.exported = []
+
+    def add(self, options: "Union[Option, List[Option]]" = []) -> None:
+        """Add new options"""
+        if isinstance(options, Option):
+            options = [options]
+        for item in options:
+            if not isinstance(item, Option):
+                raise TypeError(f"Invalid option with type '{type(item)}'")
+            self.options[item.name] = item
+
+    def list_options(self) -> str:
+        """Create formatted list of available configuration options"""
+        options = list(self.options.keys())
+
+        # create alphabetically sorted list
+        options.sort()
+        n_columns = 3
+        n_rows = len(options) // n_columns
+        n_extra = len(options) % n_columns
+        if n_extra:
+            # add one additional row to fit all options
+            n_rows += 1
+            options.extend([""] * (n_columns - n_extra))
+
+        # get width of individual columns
+        max_len = [
+            max([len(options[row + col * n_rows]) + 3
+                for row in range(n_rows)])
+            for col in range(n_columns)]
+
+        # format options as alphabetically ordered columns
+        out = []
+        for row in range(n_rows):
+            line = []
+            for col in range(n_columns):
+                line.append(f"{options[row + col * n_rows]:{max_len[col]}}")
+            out.append("".join(line))
+        return "\n".join(out)
+
+    def __getitem__(self, key: str) -> "Union[str, bool, dict]":
+        """Make class subscriptable"""
+        return self.options[key]
+
+    def __len__(self) -> int:
+        """Length corresponds to number of keys"""
+        return len(self.options)
+
+    def __iter__(self) -> "Generator":
+        """Returns itself as an iterator."""
+        for k in self.options.keys():
+            yield k
+
+    def select(self, key: str = "default") -> None:
+        """Select attribute dictionary entries corresponding to *key*"""
+        for attr, item in self.options.items():
+
+            if isinstance(item.default, dict) and key in item.default:
+                self.options[attr].default = item.default[key]
+
+    def to_scons(
+            self,
+            keys: "Union[str, List[str]]" = "",
```

Comment:
Assuming tuples are also OK, you can do:

```suggestion
            keys: "Union[str, Iterator[str]]" = "",
```

Reply:
As discussed above, this  might need to be `Sequence` instead.

Reply:
`Sequence` indeed!